compiler/damlc/tests/src/DA/Test/DamlcIntegration.hs

```diff
@@ -318,9 +318,40 @@ parseRange :: String -> Range
 parseRange s =
   case traverse readMaybe (wordsBy (`elem` (":-" :: String)) s) of
     Just [rowStart, colStart, rowEnd, colEnd] ->
-        Range
-            (Position (rowStart - 1) (colStart - 1))
-            (Position (rowEnd - 1) (colEnd - 1))
+      -- When specifying ranges:
+      --  * lines are 1-based
+      --  * columns are 0-based
+      --  * the column span is open, that is, the end column is "one
+      --    past the end" of the span.
+      --
+      -- Positions in these ranges are matched against LSP coordinates
+      -- which are 0-based in both line and column and also open. Error
+      -- messages are formatted by GHC and are 1-based in line, 0-based
+      -- in column and closed (so add one to column start, one to
+      -- column end to make them 1-based, then subtract one from column
+      -- end to convert to closed!)
+      --
+      -- 'showDiagnostics' reports ranges such that lines are 1-based,
+      -- columns are 0-based and open.
+      --
+      -- Example:
+      --   If @INFO 'range=8:13-8:47':
+      --   then the actual (LSP) range is
+      --      { _start = Position {_line = 7, _character = 13}
+      --      ,   _end = Position {_line = 7, _character = 47}}
+      --   and 'showDiagnostics' reports:
+      --     Hidden:   no
+      --     Range:    8:13-8:47
+      --     Source:   linter
+      --     Severity: DsInfo
+      --     Message:  RangeTest.daml:8:14-47: Some error message.
```

Comment:
I find it slightly confusing that the range in the `Range` field starts at `8:13` whereas the range in `Message` starts at `8:14`. Is there anything we can do about this? Does ghcide suffer from similar problems?

Reply:
Yes, it is mildly confusing, hence the detailed comment. What happens in case it's not clear is that the adjusted numbers that you see in the message result from GHCs internal formatter. I suppose if we were determined we could change `showDiagnostic` and `parseRange` to account for it. I personally don't think in light of understanding it that it's worth going to the trouble. Yes, the behavior is entirely consistent with ghcide.

Reply:
My arm could be twisted though if you think it's important. Let me know.